apps/ledger-live-mobile/src/newArch/features/FirmwareUpdate/components/UpdateBanner/UpdateBanner.test.tsx

```diff
@@ -0,0 +1,399 @@
+import * as React from "react";
+import ReactNative from "react-native";
+import { screen } from "@testing-library/react-native";
+import { render } from "@tests/test-renderer";
+import { DeviceModelId } from "@ledgerhq/devices";
+import UpdateBanner from ".";
+import { State } from "~/reducers/types";
+import { makeMockSettings } from "./__mocks__/makeMockSettings";
+
+// Mock react-navigation's useRoute and useNavigation
+jest.mock("@react-navigation/native", () => ({
+  ...jest.requireActual("@react-navigation/native"),
+  useRoute: jest.fn().mockReturnValue({ params: {} }),
+  useNavigation: jest.fn().mockReturnValue({ navigate: jest.fn() }),
+}));
+
+// Mock useLatestFirmware
+jest.mock("@ledgerhq/live-common/device/hooks/useLatestFirmware", () => ({
+  useLatestFirmware: jest.fn(),
+}));
+const { useLatestFirmware } = jest.requireMock(
+  "@ledgerhq/live-common/device/hooks/useLatestFirmware",
+);
+
+// Mock navigation to new and old update flows
+jest.mock("../../utils/navigateToNewUpdateFlow", () => ({
+  navigateToNewUpdateFlow: jest.fn(),
+}));
+jest.mock("../../utils/navigateToOldUpdateFlow", () => ({
+  navigateToOldUpdateFlow: jest.fn(),
+}));
+const { navigateToNewUpdateFlow } = jest.requireMock("../../utils/navigateToNewUpdateFlow");
+const { navigateToOldUpdateFlow } = jest.requireMock("../../utils/navigateToOldUpdateFlow");
+
+const oldUpdateFlowNotSupportedDataSet: Array<{
+  deviceModelId: DeviceModelId;
+  version: string;
+  productName: string;
+}> = [
+  { deviceModelId: DeviceModelId.nanoS, version: "1.6.0", productName: "Nano S" },
+  { deviceModelId: DeviceModelId.nanoX, version: "1.2.9", productName: "Nano X" },
+  { deviceModelId: DeviceModelId.nanoSP, version: "0.9.9", productName: "Nano S Plus" },
+];
+
+const oldUpdateFlowSupportedDataSet: Array<{
+  deviceModelId: DeviceModelId;
+  version: string;
+  productName: string;
+}> = [
+  { deviceModelId: DeviceModelId.nanoS, version: "1.6.1", productName: "Nano S" },
+  { deviceModelId: DeviceModelId.nanoX, version: "1.3.0", productName: "Nano X" },
+  { deviceModelId: DeviceModelId.nanoSP, version: "1.0.0", productName: "Nano S Plus" },
+];
```

Comment:
I think that you can reduce copy paste here, with shared TS type + default object per device that you spread and override with only the value you want to change